### PR TITLE
Add the challenge detail page

### LINKED ISSUE #4: Detail challenge page
Route `/challenge/{id}`

### PR DESCRIPTION
**Pull request by dgrcode**
Fix #4
Simple details page

### CODE REVIEW

File: packages/react-app/src/components/ChallengeCard.jsx [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+const { Text, Link: AntdLink } = Typography;
+
+const ChallengeCard = ({ challengeId, challengeInfo, submissionInfo }) => (
   <div style={{ display: "inline-block", opacity: challengeInfo.disabled ? 0.5 : 1 }}>
     {challengeInfo.disabled ? (
-      <span>{challengeInfo.label}</span>
+      <Text>{challengeInfo.label}</Text>
     ) : (
-      <a href={challengeInfo.url} target="_blank" rel="noreferrer noopener">
+      <Link to={`/challenge/${challengeId}`} component={AntdLink}>
```
Comment (carletex) [inline packages/react-app/src/components/ChallengeCard.jsx:12]: For some reaseon, having `component={AntdLink}` makes the page reload when clicking the link.

File: packages/react-app/src/views/ChallengeDetailView.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  return (
+    <div>
+      <Title>{challenge.label}</Title>
+      <Space direction="vertical">
+        <Paragraph>{challenge.description}</Paragraph>
+
+        <AntdLink href={challenge.url} target="_blank">
+          Link to challenge
+        </AntdLink>
```
Comment (carletex) [inline packages/react-app/src/views/ChallengeDetailView.jsx:30]: Maybe we can add a "< Back to challenges" link here?

File: packages/react-app/src/components/ChallengeCard.jsx (hunk 1 above)
Comment (dgrcode) [reply packages/react-app/src/components/ChallengeCard.jsx:12]: That's true. Not sure why they recommend replacing the component then... https://ant.design/components/typography/#How-to-use-Typography.Link-in-react-router

I'll take that out.

File: packages/react-app/src/views/ChallengeDetailView.jsx (hunk 2 above)
Comment (dgrcode) [reply packages/react-app/src/views/ChallengeDetailView.jsx:31]: sounds good. I've added this.

**Review by carletex — APPROVED** [review]

**Pull request merged**